core/services/subscription.go

```diff
@@ -63,52 +57,63 @@ func StartJobSubscription(job models.JobSpec, head *models.Head, store *strpkg.S
 
 	if replayFromBlock := store.Config.ReplayFromBlock(); replayFromBlock >= 0 {
 		if replayFromBlock >= nextHead.Int64() {
-			logger.Infof("StartJobSubscription: Next head was supposed to be %v but ReplayFromBlock flag manually overrides to %v, will subscribe from blocknum %v", nextHead, replayFromBlock, replayFromBlock)
+			logger.Infof("StartJobSubscription: next head was supposed to be %v but ReplayFromBlock flag manually overrides to %v, will subscribe from blocknum %v", nextHead, replayFromBlock, replayFromBlock)
 			replayFromBlockBN := big.NewInt(replayFromBlock)
 			nextHead = replayFromBlockBN
 		}
-		logger.Warnf("StartJobSubscription: ReplayFromBlock was set to %v which is older than the next head of %v, will subscribe from blocknum %v", replayFromBlock, nextHead, nextHead)
+		logger.Warnf("StartJobSubscription: replayFromBlock was set to %v which is older than the next head of %v, will subscribe from blocknum %v", replayFromBlock, nextHead, nextHead)
 	}
 
 	for _, initr := range initrs {
-		logger.Debugw("Initiator subscribing", "initr", initr)
-		unsubscriber, err := NewInitiatorSubscription(initr, store.EthClient, runManager, nextHead, store.Config, ReceiveLogRequest)
-		if err == nil {
-			unsubscribers = append(unsubscribers, unsubscriber)
-		} else {
+		filter, err := models.FilterQueryFactory(initr, nextHead, store.Config.OperatorContractAddress())
+		if err != nil {
 			merr = multierr.Append(merr, err)
+			continue
+		}
+		is, err := NewInitiatorSubscription(initr, store.EthClient, runManager, filter, store.Config.EthLogBackfillBatchSize(), ProcessLogRequest)
+		if err != nil {
+			merr = multierr.Append(merr, err)
+		} else {
+			is.Start()
+			initatorSubscriptions = append(initatorSubscriptions, *is)
```

Comment:
Why did this change?

Reply:
the only change is to include a call to is.Start() method here